<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TeamserverWindow</class>
 <widget class="QMainWindow" name="TeamserverWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1223</width>
    <height>867</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TeamserverWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* === Global Base === */
* {
    font-family: &quot;Fira Code&quot;, &quot;Consolas&quot;, monospace;
    font-size: 12pt;
    color: #c0c0c0;
}

/* === Main Window Gradient === */
QMainWindow {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 #2a2f2a, stop:1 #303d30);
}

/* === QWidget default === */
QWidget {
    background-color: #2b2f2b;
    border: none;
}

/* === QMenuBar === */
QMenuBar {
    background-color: #2b2f2b;
    color: #c0c0c0;
    font-size: 11pt;
    padding: 2px;
}
QMenuBar::item {
    padding: 2px 22px;
    font-size: 11pt;
}
QMenuBar::item:selected {
    background-color: #4d5e4d;
    color: #ffffff;
}

/* === QMenu === */
QMenu {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
}
QMenu::item:selected {
    background-color: #5a6d5a;
    color: #ffffff;
}

/* === QPushButton === */
QPushButton {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    border-radius: 5px;
    padding: 6px 12px;
}
QPushButton:hover {
    background-color: #4d5e4d;
}
QPushButton:pressed {
    background-color: #2a2f2a;
    border-color: #78a378;
}
QPushButton:disabled {
    background-color: #4d5e4d;
    color: #888;
    border-color: #444;
}

/* === QLineEdit / QTextEdit === */
QLineEdit, QTextEdit {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    border-radius: 4px;
    padding: 4px;
    selection-background-color: #5a6d5a;
    selection-color: #000;
}

/* === QTableWidget === */
QTableWidget {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    gridline-color: #495e49;
    selection-background-color: #5a6d5a;
    selection-color: #000;
}
QHeaderView::section {
    background-color: #4d5e4d;
    color: #c0c0c0;
    border: 1px solid #495e49;
    padding: 4px;
}
QTableCornerButton::section {
    background: #4d5e4d;
    border: 1px solid #495e49;
}

/* === QTabWidget === */
QTabWidget::pane {
    border: 2px solid #5a6d5a;
}
QTabBar::tab {
    background: #3a453a;
    border: 1px solid #4d5e4d;
    padding: 6px 14px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
}
QTabBar::tab:selected {
    background: #5a6d5a;
    color: #ffffff;
    font-weight: bold;
    border: 1px solid #78a378;
}

/* === QScrollBar === */
QScrollBar:vertical, QScrollBar:horizontal {
    background: #2a2f2a;
    border: none;
    width: 12px;
}
QScrollBar::handle {
    background: #5a6d5a;
    border-radius: 6px;
}
QScrollBar::handle:hover {
    background: #78a378;
}
QScrollBar::add-line, QScrollBar::sub-line {
    background: none;
    border: none;
}

/* === QComboBox === */
QComboBox {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    border-radius: 4px;
    padding: 4px;
}
QComboBox::drop-down {
    border-left: 1px solid #495e49;
    background: #4d5e4d;
}
QComboBox QAbstractItemView {
    background-color: #4d5e4d;
    border: 1px solid #5a6d5a;
    selection-background-color: #5a6d5a;
    selection-color: #000;
}

/* === QListWidget / QTreeView === */
QListWidget, QTreeView {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
}
QListWidget::item:selected, QTreeView::item:selected {
    background-color: #5a6d5a;
    color: #000;
    font-weight: bold;
}

/* === QSpinBox === */
QSpinBox {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    border-radius: 4px;
    padding-right: 20px;
    color: #c0c0c0;
}
QSpinBox::up-button, QSpinBox::down-button {
    background-color: #3b453b;
    border: 1px solid #5a6d5a;
    width: 16px;
    subcontrol-origin: border;
}
QSpinBox::up-button {
    subcontrol-position: top right;
    border-top-right-radius: 4px;
}
QSpinBox::down-button {
    subcontrol-position: bottom right;
    border-bottom-right-radius: 4px;
}
QSpinBox::up-button:hover, QSpinBox::down-button:hover {
    background-color: #4d5e4d;
}

/* === QLabel === */
QLabel {
    color: #c0c0c0;
}

/* === ToolTip === */
QToolTip {
    background-color: #5a6d5a;
    color: #000;
    border: 1px solid #78a378;
    padding: 4px;
    border-radius: 4px;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" columnstretch="17,3,3">
    <property name="leftMargin">
     <number>15</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>15</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <property name="horizontalSpacing">
     <number>25</number>
    </property>
    <property name="verticalSpacing">
     <number>10</number>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_2">
      <property name="font">
       <font>
        <family>Fira Code</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Agents</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QLabel" name="label_3">
      <property name="font">
       <font>
        <family>Fira Code</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Listeners</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="0" rowspan="2">
     <widget class="AgentTableWidget" name="agentTable">
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="tabKeyNavigation">
       <bool>false</bool>
      </property>
      <property name="showDropIndicator" stdset="0">
       <bool>false</bool>
      </property>
      <property name="dragDropOverwriteMode">
       <bool>false</bool>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <property name="showGrid">
       <bool>true</bool>
      </property>
      <property name="gridStyle">
       <enum>Qt::PenStyle::NoPen</enum>
      </property>
      <property name="rowCount">
       <number>0</number>
      </property>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>50</number>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Hostname</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>User</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Process</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>IP</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Last Seen (s)</string>
       </property>
      </column>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="ListenerTableWidget" name="listenersTable">
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="tabKeyNavigation">
       <bool>false</bool>
      </property>
      <property name="showDropIndicator" stdset="0">
       <bool>false</bool>
      </property>
      <property name="dragDropOverwriteMode">
       <bool>false</bool>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <property name="gridStyle">
       <enum>Qt::PenStyle::NoPen</enum>
      </property>
      <property name="rowCount">
       <number>0</number>
      </property>
      <property name="columnCount">
       <number>3</number>
      </property>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>50</number>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>50</number>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderDefaultSectionSize">
       <number>35</number>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Interface</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Port</string>
       </property>
      </column>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="createListenerButton">
      <property name="text">
       <string>Create</string>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="deleteListenerButton">
      <property name="text">
       <string>Delete</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="agentLabel">
      <property name="font">
       <font>
        <family>Fira Code</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Sunken</enum>
      </property>
      <property name="text">
       <string>Agent : None</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="1" colspan="2">
     <widget class="QLabel" name="label_4">
      <property name="font">
       <font>
        <family>Fira Code</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Activity Log</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QTextEdit" name="cmdsTextEdit">
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Fira Code','Consolas','monospace'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Fira Code','Consolas','Courier New','monospace'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1" rowspan="2" colspan="2">
     <widget class="QListWidget" name="activityList">
      <property name="isWrapping" stdset="0">
       <bool>false</bool>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLineEdit" name="commandLine">
      <property name="maxLength">
       <number>999999999</number>
      </property>
      <property name="frame">
       <bool>true</bool>
      </property>
      <property name="placeholderText">
       <string>Enter a command...</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1223</width>
     <height>27</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menu_Strata_C2">
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Strata C2</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuPayloads">
    <property name="title">
     <string>Payloads</string>
    </property>
    <addaction name="actionGenerate"/>
   </widget>
   <widget class="QMenu" name="menuArmory">
    <property name="title">
     <string>Armory</string>
    </property>
    <addaction name="actionImport_Assembly"/>
   </widget>
   <addaction name="menu_Strata_C2"/>
   <addaction name="menuPayloads"/>
   <addaction name="menuArmory"/>
  </widget>
  <action name="actionImport_Assembly">
   <property name="text">
    <string>Import Assembly...</string>
   </property>
  </action>
  <action name="actionGenerate">
   <property name="text">
    <string>Generate...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ListenerTableWidget</class>
   <extends>QTableWidget</extends>
   <header>../../../components/TableWidgets.h</header>
  </customwidget>
  <customwidget>
   <class>AgentTableWidget</class>
   <extends>QTableWidget</extends>
   <header>../../../components/TableWidgets.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
